Plutarch/Builtin.hs

```diff
@@ -1,27 +1,122 @@
-module Plutarch.Builtin (PData (..), pfstBuiltin, psndBuiltin, pasConstr, PBuiltinPair, PBuiltinList) where
+module Plutarch.Builtin (
+  PData (..),
+  pTrace,
+  (#£),
+  (!£),
+  PBuiltin (..),
+  PList (..),
+  PPair (..),
+) where
 
-import Plutarch (punsafeBuiltin)
+import Data.Proxy
+import Data.Text (Text)
+import Data.Type.Nat
+import Plutarch
+import Plutarch.Bool
+import Plutarch.Builtin.List.Type
+import Plutarch.Builtin.Pair.Type
 import Plutarch.ByteString (PByteString)
 import Plutarch.Integer (PInteger)
 import Plutarch.Prelude
+import Plutarch.String
 import qualified PlutusCore as PLC
 
-data PBuiltinPair (a :: k -> Type) (b :: k -> Type) (s :: k)
```

Comment:
Please don't change the name nor remove this. We want a separate `PPair` eventually.